src/cpp/src/llm_pipeline.cpp

```diff
@@ -222,77 +214,98 @@ class StatefulLLMPipeline final : public LLMPipelineImplBase {
                         "(input_ids, attention_mask, position_ids, beam_idx) "
                         "but you have '" + std::to_string(num_inputs) + "' inputs");
 
+        if (is_chat_conversation) {
+            std::copy(input_ids.data<int64_t>(), input_ids.data<int64_t>() + input_ids.get_size(), std::back_inserter(m_tokenized_chat_history));
+        }
+        ov::Tensor tokenized_chat_history = ov::Tensor(ov::element::i64, {1, m_tokenized_chat_history.size()}, m_tokenized_chat_history.data());
+        bool is_cache_empty = !m_selected_beam.has_value();
```

Comment:
as I see `m_selected_beam` is either 0 or nullopt, so it emulates `bool`

maybe `m_is_cache_empty`  is better name then ? because here we convert `m_selected_beam` to bool anyway

Reply:
moved to m_is_cache_empty